syntax = "proto3";

service NLPService {
  rpc MatchScenario (MatchScenarioRequest) returns (MatchScenarioResponse);
  rpc ExtractArguments (ExtractArgumentsRequest) returns (ExtractArgumentsResponse);
}

message MatchScenarioRequest {
  string user_prompt = 1;
}

message MatchScenarioResponse {
  string scenario_name = 1;
  string root_id = 2;
}

message ExtractArgumentsRequest {
  string user_prompt = 1;
}

message Argument {
  string value = 1;
  string type = 2;
}

message ExtractArgumentsResponse {
  map<string, string> arguments = 1;
}
